gateway/hydrator.go

```diff
@@ -39,13 +40,17 @@ var endpointHydratorRunInterval = 10_000 * time.Millisecond
 // 3. Reporting the results back to the service's QoS instance.
 type EndpointHydrator struct {
 	Protocol
-	QoSPublisher
 
-	// ServiceQoSGenerators provides the hydrator with the EndpointCheckGenerator
-	// it needs to invoke for a service ID.
-	// ServiceQoSGenerators should not be modified after the hydrator is started.
-	ServiceQoSGenerators map[protocol.ServiceID]QoSEndpointCheckGenerator
-	Logger               polylog.Logger
+	// ActiveQoSService provides the hydrator with the QoS instances
+	// it needs to invoke for generating synthetic service requests.
+	// ActiveQoSService should not be modified after the hydrator is started.
+	ActiveQoSService map[protocol.ServiceID]QoSService
+	Logger           polylog.Logger
+
+	// MetricsReporter and DataReporter are intentionally declared separately, rather than using a slice of the same interface, to be consistent
```

Comment:
Ditto my other comment about what to include in the comments.

Also, do you think it'd be overkill to have a `Reporters` struct IFF we anticipate to ALWAYS require both?